contracts/cw-core/src/contract.rs

```diff
@@ -181,19 +189,67 @@ pub fn execute_pause(
         .add_attribute("until", until.to_string()))
 }
 
+pub fn execute_admin_msgs(
+    deps: Deps,
+    sender: Addr,
+    msgs: Vec<CosmosMsg<Empty>>,
+) -> Result<Response, ContractError> {
+    let admin = ADMIN.may_load(deps.storage)?;
+
+    match admin {
+        Some(admin) => {
+            // Check if the sender is the DAO Admin
+            if sender != admin {
+                return Err(ContractError::Unauthorized {});
+            }
+
+            Ok(Response::default()
+                .add_attribute("action", "execute_admin_msgs")
+                .add_messages(msgs))
+        }
+        None => Err(ContractError::NoAdmin {}),
+    }
+}
+
 pub fn execute_proposal_hook(
     deps: Deps,
     sender: Addr,
     msgs: Vec<CosmosMsg<Empty>>,
 ) -> Result<Response, ContractError> {
+    // Check that the message has come from one of the proposal modules
     if !PROPOSAL_MODULES.has(deps.storage, sender) {
         return Err(ContractError::Unauthorized {});
     }
+
     Ok(Response::default()
         .add_attribute("action", "execute_proposal_hook")
         .add_messages(msgs))
 }
 
+pub fn execute_update_admin(
+    deps: DepsMut,
+    sender: Addr,
+    admin: Option<Addr>,
+) -> Result<Response, ContractError> {
+    let current_admin = ADMIN.load(deps.storage)?;
```

Comment:
Might be nice to use `may_load` and return `ContractError::NoAdmin` here. Otherwise the error seems liable to be pretty unreadable.